src/server/scripts/Spells/spell_druid.cpp

```diff
@@ -922,7 +922,23 @@ class spell_dru_starfall_dummy : public SpellScript
 
     void FilterTargets(std::list<WorldObject*>& targets)
     {
+        // Get caster object
+        Unit* caster = GetCaster();
+        if (!caster)
+            return; // If caster is null, exit early
+        // Remove targets if they are outside line of sight with respect to caster
+        targets.remove_if([caster](WorldObject const* target)
+            {
+                if (Unit const* unitTarget = target->ToUnit())
```

Comment:
```suggestion
                if (target)
```
Is there a reason this isn't an option?

Reply:
Agreed, this also made the check fail. I will correct